src/utils/database-common/src/transactions/db_transaction_manager.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::any::Any;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use dill::{Catalog, CatalogBuilder};
+use internal_error::{InternalError, ResultIntoInternal};
+use tokio::sync::Mutex;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait DatabaseTransactionManager: Send + Sync {
+    async fn make_lazy_transaction_ref(&self) -> Result<LazyTransactionRef, InternalError>;
```

Comment:
Minor: You could accept `Arc<Self>` here instead of `&self` to avoid allocating a `Box` for a copy of the connection pool per every transaction.

Reply:
Cloning pool is cheap, it's an `Arc`. 
We have to allocate another `Arc` and `Mutex` anyway.
Don't see how this suggestion improves anything.